packages/website/docs/docs/block-types.md

```diff
@@ -102,4 +102,244 @@ type DefaultBlockProps = {
+
+#### `render`
+
+This is a React component which defines how your custom block should be rendered in the editor, and takes two props:
+
+`block:` The block that should be rendered.
+
+`editor:` The BlockNote editor instance that the block is in.
+
+For our custom image block, we use a parent `div` which contains the image and caption. Since `block` will always be an `image` block, we also know it contains a `src` prop, and can pass it to the child `img` element.
+
+But what's this `InlineContent` component? Since we set `containsInlineContent` to `true`, it means we want to include an editable rich-text field somewhere in the image block. You should use the `InlineContent` component to represent this field in your `render` component. Since we're using it to create our caption, we add it below the `img` element.
+
+While the `InlineContent` component can be put anywhere inside the component you pass to `render`, you should make sure to only have one. If `containsInlineContent` is set to false, `render` shouldn't contain any.
+
+### Adding Custom Blocks to the Editor
+
+After creating `BlockConfig`s for all of our custom blocks, we need to convert them into `BlockSpec`s. A `BlockSpec` just tells BlockNote how your custom block should be defined internally, so we don't need to know much about it. We just need to convert our `BlockConfig` to a `BlockSpec` using the `createReactBlockSpec` function:
+
+```typescript jsx
+function createReactBlockSpec(blockConfig: BlockConfig): BlockSpec {...};
+```
+
+Now, all we need to do is pass it to the editor using the `blockSchema` option, which tells BlockNote which blocks to use. Let's again look at the image block from the demo as an example:
+
+```typescript jsx
+const ImageBlock = createReactBlockSpec(ImageBlockConfig);
+
+...
+
+// Creates a new editor instance.
+const editor: BlockNoteEditor<
+  DefaultBlockSchema & { image: typeof ImageBlock }
+> | null = useBlockNote({
+  // Tells BlockNote which blocks to use.
+  blockSchema: {
+    // Adds all default blocks.
+    ...defaultBlockSchema,
+    // Adds the custom image block.
+    image: ImageBlock,
+  }
+})
+```
+
+Notice three details about this code snippet. First, since we still want the editor to use the [Built-In Block Types](/docs/block-types#built-in-block-types), we add `defaultBlockSchema` to our custom block schema. Second, the key which we use for the custom image block is the same string we use for its type. Make sure that this is always the case for your own custom blocks. Finally, we provide the type of our schema as a type argument to `BlockNoteEditor`.
+
+And we're done! You now know how to create custom blocks and add them to the editor. Head to [Manipulating Blocks](/docs/manipulating-blocks) to see what you can do with them in the editor.
+
+### Limitations
```

Comment:
Let's replace the entire Limitations (including the notes on multiple rich text fields and contenteditable) to an "alert"

> While custom blocks open a lot of doors for what you can do with BlockNote, we're still working on improving the API and there are a few limitations for the kinds of blocks you can create.
> We'd love to hear your feedback on Github or in our Discord community!

Reasoning: we don't know yet what people will run into / which use-cases they envision. The below examples might only scare them away